docs/guillotine.adoc

```diff
@@ -0,0 +1,1349 @@
+<2> A content type for a single _movie_,
+<3> A part with a controller that fetches child content items of the _movie_ content type, and renders them into MovieList,
+<4> In _site.xml_ we will set up controller mappings for both the guillotine API and...
+<5> ...the single-movie preview controller: displays a single movie without needing to set up a template and a part.
+<6> _guillotineApi.es6_ is the actual API to guillotine. It can run graphQL queries both from XP controllers and through received HTTP requests. And _guillotineRequest.es6_ simplifies making such a request from the browser. Both of these are general-purpose and come with the starter (since version 1.1.0). But _helpers/movieListRequests.es6_ contains helper functions specific to the lesson site we're building here: it helps with building a query for fetching movie-list data, and parsing the returned data into the `props` format that the _Movie_ component needs. These helpers are also used on both frontend and backend.
+
+
+{zwsp} +
+{zwsp} +
+{zwsp} +
+
+
+== Groundwork: movie items
+
+This first stage should be easy enough, almost entirely repeating steps you've been through in previous chapters. We'll make a _movie_ content type, set up react4xp to preview-render it with react components (but with <<#controller_mapping, a little twist>>), and add some movie items that will be listed when the site is done.
+
+[NOTE]
+====
+This entire chapter builds on the <<imports-and-dependency-chunks#webpack_config, config setup from the previous lesson>>: _react4xp.properties_, _webpack.config.react4xp.js_ and the extra NPM packages should be set up like that.
+
+If you haven't completed that section already, better take a couple of minutes and do that before proceeding.
+====
+
+{zwsp} +
+
+=== Movie content type
+
+When the setup is ready, we'll start by adding a _movie_ *content type*, with an ImageSelector for a poster `image`, a simple HtmlArea with a movie `description`, a numeral Long field for adding the release `year` and an array of `actor` names:
+
+.site/content-types/movie/movie.xml:
+[source,xml,options="nowrap"]
+----
+<content-type xmlns="urn:enonic:xp:model:1.0">
+  <display-name>Movie</display-name>
+  <description>Moving images often reflecting culture</description>
+  <super-type>base:structured</super-type>
+
+  <form>
+    <input name="image" type="ImageSelector">
+        <label>Movie poster</label>
+        <occurrences minimum="1" maximum="1"/>
+    </input>
+
+    <input name="description" type="HtmlArea">
```

Comment:
What did you mean, it should be a text area (or that it practically is)?

I just thought it'd be neat to both allow bold/italic/underline, which requires a html area (and demonstrates that transferring unescaped data and rendering with dangerouslyInsertHtml works). 

It's not really a big deal, of course I can fix it if you think it makes things clearer.

Reply:
This practically is a text area. I just don't understand why you excluded 90% of the html features. Just allow all no need to restrict on a demo.

Reply:
Excluding those features enables me to skip the processHtml step in XP...

Reply:
How about you @sigdestad , do you have an opinion? Between:
1. using a textarea and keeping it real simple,
2. like it is now: a _simplifiled_ htmlarea with only bold/italic/underline, so that it processHtml isn't needed and we keep the server side of things simpler... but still have one content point that delivers html tags so that it demonstrates an instance of unescaped rendering (`dangerouslyInsertHtml` = `data-th-utext`) still working in a headless context, OR
3. allowing full html features, so that the entire flow of that is demonstrated in a headless/hybrid context? On the downside: increases the complexity of the lesson a bit, and will require more writing/polish on these docs

Reply:
❓

Reply:
-> https://github.com/enonic/doc-react4xp/issues/7

Reply:
And fix for now: solution 1.